packages/core/src/Serialization/Certificates/PoolRegistration.ts

```diff
@@ -13,6 +13,7 @@ const EMBEDDED_GROUP_SIZE = 10;
  * about the pool such as the pledge, costs, margin, reward account, and the pool's owners and relays.
  */
 export class PoolRegistration {
+  #deposit: Cardano.Lovelace;
```

Comment:
I am not sure we should add this here 🤔 . This class should be a representation of the CBOR pool registration certificate and it does not have this field. This means that if you serialize and deserialize to and from CBOR this information will be lost, which is unexpected. IMHO this class should be kept as is without the field and default to 0 when converting to the core type.

This field is only relevant in the core type when we get the information from the backend, but not when we read from CBOR as the information is not there, another option would be to make the field optional in the core type?

Reply:
Crystal clear why this property shouldn't be here: I'll remove it.
While I didn't understood why you are proposing to make the field optional in the core type; could you please elaborate?
Thank you

Reply:
This class maps to its equivalent core type (toCore), but doesn't have this information (the deposit), so we need to either default the value to 0, or make the field optional so the mapping can occur

Reply:
Much more clear; thank you.
I'll dig this aspect a bit more.

Reply:
f2c51e9e8e should solve this